README.md

```diff
@@ -352,6 +352,23 @@ generic_terminal.talon
 
 - New programming languages should support the appropriate 'generic' grammars where possible, see above.
 
+## Automatic formatting/linters
+
+This repository uses [`pre-commit`](https://pre-commit.com/) to run manage its formatters/linters. To set up `pre-commit`:
+
+```bash
+$ pip install pre-commit
+```
+
+You then have a few options as to when to run it:
+
+- Run at any time, on your locally changed files: `pre-commit run`
+- Run on all files in the repository: `pre-commit run --all-files`
+- Install a git pre-commit hook, so `pre-commit` runs automatically before creating local commits: `pre-commit install`, and will block then if the formatting is wrong.
+- Run automatically on your PR branches (fixes will be pushed automatically): Visit https://pre-commit.ci/ and authorize the app to connect to your knausj repo(s).
```

Comment:
I don't believe this is necessary for those creating PRs? Should be enough to allow it write access to the main repo, and PR authors can allow write access by maintainers.

Reply:
Oh, interesting, I hadn't considered this. So by combining the facts that:

1) authorizing the service in the original repo makes it a contributor
2) fork creators can allow contributors of the original upstream repo to modify their PRs

...contributors from forked repos won't need to add the service if they check the box for #2?

If so very cool.

Reply:
(Done)